File: astera/2/euler2.rb
n = 4000000

a = 1
b = 2
fibSequ = [1, 2]
sum = 0

while (b + a) <= n do
    a = b + a
    fibSequ.push(a)
    
    if (a + b) <= n
        b = a + b
        fibSequ.push(b)
    end
end
puts fibSequ

fibSequ.each do |i|
    if i % 2 == 0
        sum = sum + i
    end
end

puts "Sum = #{sum}"

File: astera/2/euler2.c
#include <stdio.h>

int main()
{
    int n = 4000000;
    int a = 1;
    int b = 2;
    int sum = 2;
    
    for(sum = 2; sum < n;)
    {
        a = b + a;
        b = a + b;
        if(a % 2 == 0)
        {
            sum += a;
        }
        if(b % 2 == 0)
        {
            sum += b;
        }
    }
    printf("Sum = %d\n", sum);
}
